cedar-policy/src/api.rs

```diff
@@ -1762,16 +1762,16 @@ impl Diagnostic for ValidationResult {
 /// and provides details specific to that kind of problem. The error also records
 /// where the problem was encountered.
 #[derive(Debug, Clone, Error)]
-#[error("validation error on {location}: {}", self.error_kind())]
+#[error("{error}")]
 pub struct ValidationError {
     location: SourceLocation,
```

Comment:
Do we still need this `location` field, if there's now a `location` embedded in the `cedar_policy_validator::ValidationError`?  Looks like in the `Diagnostic` impl for this type we're always deferring to the location in `error`

Reply:
The `source_location()` function here needed to return a reference. Changed `SourceLocation` to be `RefCast` so that it can wrap the internal reference type.

Reply:
Looks good